commons-numbers-complex-arrays/src/test/java/org/apache/commons/numbers/complex/arrays/ComplexListTest.java

```diff
@@ -137,92 +256,119 @@ void testSetAddAndAddAllNullPointerException() {
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(-2, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(size, Complex.ofCartesian(200, 1)));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.set(size + 1, Complex.ofCartesian(200, 1)));
     }
 
-    @Test
-    void testAddIndexOutOfBoundExceptions() {
-        ComplexList list = ComplexList.interleaved();
+    @ParameterizedTest
+    @MethodSource({"listImplementations"})
+    void testAddIndexOutOfBoundExceptions(List<Complex> l) {
         int size = 5;
-        IntStream.range(0, size).mapToObj(i -> Complex.ofCartesian(i, -i)).forEach(list::add);
+        IntStream.range(0, size).mapToObj(i -> Complex.ofCartesian(i, -i)).forEach(l::add);
 
         Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
-            list.add(-1, Complex.ofCartesian(42, 13)));
+            l.add(-1, Complex.ofCartesian(42, 13)));
         Assertions.assertThrows(IndexOutOfBoundsException.class, () ->
-            list.add(size + 1, Complex.ofCartesian(42, 13)));
+            l.add(size + 1, Complex.ofCartesian(42, 13)));
     }
 
-    @Test
-    void testRemoveIndexOutOfBoundExceptions() {
-        ComplexList list = generateComplexList(2);
-        list.remove(0);
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(1));
-        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> list.remove(-1));
+    @ParameterizedTest
+    @MethodSource({"listImplementationsWithSize"})
+    void testRemoveIndexOutOfBoundExceptions(List<Complex> l, int size) {
+        IntStream.range(0, size).forEach(i -> l.add(Complex.ofCartesian(i, -i)));
+        l.remove(0);
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.remove(1));
+        Assertions.assertThrows(IndexOutOfBoundsException.class, () -> l.remove(-1));
     }
 
     @ParameterizedTest
     @ValueSource(ints = {0, 1, 10})
     void testConstructor(int size) {
         List<Complex> l1 = new ArrayList<>(size);
         List<Complex> l2 = ComplexList.interleaved(size);
+        List<Complex> l3 = new ArrayList<>(size);
+        List<Complex> l4 = ComplexList.nonInterleaved(size);
+
         Assertions.assertEquals(l1, l2);
+        Assertions.assertEquals(l3, l4);
+
         assertListOperation(l -> l.add(Complex.ofCartesian(10, 20)), l1, l2);
         assertListOperation(l -> {
             l.add(1, Complex.ofCartesian(10, 20));
             return Boolean.TRUE;
         }, l1, l2);
         assertListOperation(l -> l.addAll(1, l), l1, l2);
+
+        assertListOperation(l -> l.add(Complex.ofCartesian(10, 20)), l3, l4);
+        assertListOperation(l -> {
+            l.add(1, Complex.ofCartesian(10, 20));
+            return Boolean.TRUE;
+        }, l3, l4);
+        assertListOperation(l -> l.addAll(1, l), l3, l4);
     }
 
     @Test
     void testCapacityExceptions() {
```

Comment:
Again this could be parameterized with the list constructor and an int containing the max capacity.

Note if you use the `@MethodSource` annotation without any arguments then the static method is assumed to be the same name as the test method. So you can have:
```Java
    static Stream<Arguments> testCapacityExceptions() {
        return Stream.of(
            Arguments.of((IntFunction<List<Complex>>) ComplexList::interleaved, MAX_CAPACITY_INTERLEAVED),
            Arguments.of((IntFunction<List<Complex>>) ComplexList::nonInterleaved, MAX_CAPACITY_NON_INTERLEAVED)
        );
    }

    @ParameterizedTest
    @MethodSource
    void testCapacityExceptions(IntFunction<List<Complex>> constructor, int maxCapacity) {
        Assertions.assertThrows(IllegalArgumentException.class, () -> constructor.apply(maxCapacity + 1));

```